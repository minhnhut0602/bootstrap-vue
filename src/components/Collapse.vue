<template>
    <div
        class="d-inline-block"
        data-bs-toggle="collapse"
        :data-bs-target="`#${id}`"
    >
        <slot
            name="toggle"
        />
    </div>

    <div
        :id="id"
        class="collapse"
    >
        <slot />
    </div>
</template>

<script lang="ts" setup>
import {defineProps} from 'vue';
import {idProps} from '@/composables/useId';

defineProps({
    ...idProps,
});
</script>
