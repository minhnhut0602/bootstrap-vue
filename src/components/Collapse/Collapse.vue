<template>
    <div
        class="d-inline-block"
        data-bs-toggle="collapse"
        :data-bs-target="`#${id}`"
        @click="clickToggle"
    >
        <slot
            name="toggle"
        />
    </div>

    <div
        :id="id"
        class="collapse"
        :data-bs-parent="parentSelector"
    >
        <slot />
    </div>
</template>

<script lang="ts">
import {idProps} from '@/composables/useId';
</script>

<script lang="ts" setup>
defineProps({
    ...idProps,
    // Should be a valid query selector.
    parentSelector: {
        type: String,
        default: null,
    },
});

const emit = defineEmits(['click:toggle']);

const clickToggle = () => emit('click:toggle');
</script>
