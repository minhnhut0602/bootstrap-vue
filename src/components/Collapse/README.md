# Collapse

```vue
<Collapse parent-selector="#parent">
    <template #toggle-content="{id}">
        <Button
            data-bs-toggle="collapse"
            :data-bs-target="'#' + id"
        >
            Open me
        </Button>
    </template>

    <Card text="This panel is hidden by default but revealed when the user activates the relevant trigger." />
</Collapse>
```

## Props

| Prop name      | Description                                                                    | Type   | Values | Default |
| -------------- | ------------------------------------------------------------------------------ | ------ | ------ | ------- |
| parentSelector | The selector of the parent element<br/>_This should be a valid query selector_ | string | -      | null    |

## Slots

| Name          | Description                          | Bindings                                        |
| ------------- | ------------------------------------ | ----------------------------------------------- |
| toggle        | Displays the toggle for the collapse | **id** `string` - The id of the target collapse |
| toggleContent | Displays the content for the toggle  | **id** `string` - The id of the target collapse |
| default       | Displays the content of the collapse | **id** `string` - The id of the collapse        |
