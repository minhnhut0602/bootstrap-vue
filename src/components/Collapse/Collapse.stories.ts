import Btn from '@/components/Button';
import Card from '@/components/Card';
import Collapse from '.';

export default {
    title: 'Components/Collapse',
    component: Collapse,
};

const template = (args) => ({
    components: {Btn, Card, Collapse},
    setup: () => ({args}),
    template: `
        <Collapse v-bind="args">
            <template #toggle>
                <Btn class="mb-2">Open me</Btn>
            </template>

            <Card text="Some placeholder content for the collapse component. This panel is hidden by default but revealed when the user activates the relevant trigger." />
        </Collapse>
    `,
});

export const Default = template.bind({});

