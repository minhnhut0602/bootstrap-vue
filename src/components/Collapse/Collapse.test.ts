import Collapse from '.';
import {shallowMount} from '@vue/test-utils';

describe('template', () => {
    componentRenderTest(Collapse, {
        // Force an id to prevent snapshots from failing because of uuid.
        props: {id: 'dont-change-please'},
    });

    componentSlotRenderTest(Collapse);

    it('emits event on click collapse toggle', async() => {
        const wrapper = shallowMount(Collapse);

        wrapper.findComponent(Collapse).vm.$emit('click:toggle');

        expect(wrapper.emitted('click:toggle')).toBeTruthy();
    });
});
