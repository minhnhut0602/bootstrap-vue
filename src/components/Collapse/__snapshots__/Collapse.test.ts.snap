// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`template renders default 1`] = `
"<!--
    @slot Displays the toggle for the collapse
    @binding {string} id The id of the target collapse
    -->
<div class=\\"collapse-toggle\\" data-bs-toggle=\\"collapse\\" data-bs-target=\\"#dont-change-please\\">
  <!--
            @slot Displays the content for the toggle
            @binding {string} id The id of the target collapse
            -->
</div>
<div id=\\"dont-change-please\\" class=\\"collapse\\">
  <!--
        @slot Displays the content of the collapse
        @binding {string} id The id of the collapse
        -->
</div>"
`;
